.golangci.yml

```diff
@@ -9,23 +9,105 @@ linters-settings:
   gofmt:
     # simplify code: gofmt with `-s` option, true by default
     simplify: true
+  tagliatelle:
+    case:
+      rules:
+        json: snake
+  whitespace:
+    multi-func: true
+    multi-if: true
+  gosec:
+    excludes:
+      - G402 # Look for bad TLS connection settings.
+      - G306 # Poor file permissions used when writing to a new file.
+  staticcheck:
+    go: "1.18"
+    checks: ["-SA1019"]
 
 linters:
   enable-all: true
   disable:
-    # Init functions are used by loggers throughout the codebase.
-    - gochecknoinits
-
-    # Global variables are used by loggers.
+    # Global variables are used in many places throughout the code base.
     - gochecknoglobals
 
-    # Some lines are over 80 characters on purpose and we don't want to make
-    # them even longer by marking them as 'nolint'.
+    # Some lines are over 80 characters on purpose and we don't want to make them
+    # even longer by marking them as 'nolint'.
     - lll
 
-    # We don't care (enough) about misaligned structs to lint that.
-    - maligned
+    # We want to allow short variable names.
+    - varnamelen
 
-    # We have long functions, especially in tests. Moving or renaming those
-    # would trigger funlen problems that we may not want to solve at that time.
+    # We want to allow TODOs.
+    - godox
+    
+    # We have long functions, especially in tests. Moving or renaming those would
+    # trigger funlen problems that we may not want to solve at that time.
     - funlen
+    
+    # Disable for now as we haven't yet tuned the sensitivity to our codebase
+    # yet. Enabling by default for example, would also force new contributors to
+    # potentially extensively refactor code, when they want to smaller change to
+    # land.
+    - gocyclo
+    - gocognit
+    - cyclop
+    
+    # Instances of table driven tests that don't pre-allocate shouldn't trigger
+    # the linter.
+    - prealloc
+    
+    # Init functions are used by loggers throughout the codebase.
+    - gochecknoinits
+    
+    # Causes stack overflow, see https://github.com/polyfloyd/go-errorlint/issues/19.
+    - errorlint
+    
+    # Deprecated linters. See https://golangci-lint.run/usage/linters/.
+    - interfacer
+    - golint
+    - maligned
+    - scopelint
+    
+    # New linters that need a code adjustment first.
+    - wrapcheck
+    - nolintlint
+    - paralleltest
+    - tparallel
+    - testpackage
+    - gofumpt
+    - gomoddirectives
+    - ireturn
+    - maintidx
+    - nlreturn
+    - dogsled
+    - gci
+    - containedctx
+    - contextcheck
+    - errname
+    - exhaustivestruct
+    - exhaustruct
+    - goerr113
+    - gomnd
+    - ifshort
+    - noctx
+    - nestif
+    - wsl
+    - exhaustive
+    - forcetypeassert
+    - nilerr
+    - nilnil
+    - stylecheck
+    - thelper
+    - nosprintfhostport
+    - errchkjson
```

Comment:
Didn't check with other linters, but `errchkjson` and `nosprintfhostport` could be removed and the linter still runs through just well. At some point we might need to go through all the linters and either code-adjust or change the comment above.

Reply:
Yes, I agree, we should probably take the time to set a fixed list of linters we want to have active and then synchronize all the linter setting files across all our projects.